baseplate/__init__.py

```diff
@@ -670,6 +670,20 @@ def make_child(
         """Return a child Span whose parent is this Span."""
         raise NotImplementedError
 
+    def with_tags(self, tags: Dict[str, Any]) -> "Span":
+        """This allows us to set tags in a Span before starting it in a context manager.
```

Comment:
Please write documentation in an authoritative voice:
```suggestion
        """Declare a set of tags to be added to a span before starting it in the context manager.
```